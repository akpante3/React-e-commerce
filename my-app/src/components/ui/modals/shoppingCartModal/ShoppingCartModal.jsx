/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */
import React, { Fragment } from 'react';
import { withRouter } from 'react-router-dom';
import { CSSTransition } from 'react-transition-group';
import PropsTypes from 'prop-types';
import './ShoppingCartModal.scss';
import AppButton from '../../buttons/AppButton/AppButton';


const MainLayout = ({ children, toggleCartModal, history }) => {
  const handleCartClick = () => {
    toggleCartModal();
  };

  const buttonClickHandler = (value) => {
    if (value === 'Back to shop') {
      history.push('/');
      toggleCartModal();
    } else {
      history.push('/checkout');
      toggleCartModal();
    }
  };

  return (
    <CSSTransition
      in
      appear
      timeout={2000}
      classNames="enter"
    >
      <Fragment>
        <div className="product-modal">
          <div className="product-modal-content">
            <div className="close-modal">
              <div
                className="text-right close-modal-icon"
                onClick={() => handleCartClick()}
              >
            &times;
              </div>
            </div>
            <div className="product-modal-children">
              {children}
            </div>
            <div className="row bottom-section">
              <div className="col">
                <AppButton
                  title='Back to shop'
                  value='Back to shop'
                  color='white'
                  buttonClickHandler={buttonClickHandler}
                />
              </div>
              <div className="checkout">
                <AppButton
                  title="checkout"
                  value="checkout"
                  buttonClickHandler={buttonClickHandler}
                />
              </div>
            </div>
          </div>
        </div>
        <div />
      </Fragment>
    </CSSTransition>
  );
};

MainLayout.propTypes = {
  children: PropsTypes.element.isRequired,
};

export default withRouter(MainLayout);
