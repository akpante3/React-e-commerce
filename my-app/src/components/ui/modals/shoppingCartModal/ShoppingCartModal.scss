
.product-modal {
  position: fixed;
  z-index: 3;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(255, 255, 255, 0);
  display: flex;
  justify-content: center;
  align-items: center;
  opacity: 1;
  transition: opacity 0.2s ease;
}
.close-modal {
  font-size: 19px;
  width:100%;
  padding: 3px 20px;
  display: block;
  height: 30px;
  .close-modal-icon {
    font-weight: 200;
    font-size: 40px;
    cursor: pointer;
  }
}
.product-modal-content {
  width: 60%;
  background-color: white;
  box-shadow: 1px 1px 30px 5px rgba(94, 94, 94, 0.4);
  border-radius: 5px;
  .bottom-section {
    background-color: rgb(240, 240, 240);
    padding: 14px;
    width: 100%;
    margin: 0;
  }
  .product-modal-children {
    padding: 25px;
  }
}
// transition
.enter-appear {
  opacity: 0;
  z-index: 1;
  transform: scale(0.7);
}
.enter-appear.enter-appear-active {
  opacity: 1;
  transform: translateX(0);
  transition: opacity 1000ms, transform 1000ms;
}