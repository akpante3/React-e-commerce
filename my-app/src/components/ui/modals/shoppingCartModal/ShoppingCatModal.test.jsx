import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import ShoppingCartModal from './ShoppingCartModal';

const initialState = {
  shoppingCart: { cartItems: [{}] },
};
const mockStore = configureStore();
let wrapper;
let store;

describe('test for ShoppingCartModal', () => {
  it('test for ShoppingCartModal', () => {
    store = mockStore(initialState);
    wrapper = shallow(<ShoppingCartModal store={store} />);
    expect(wrapper).toMatchSnapshot();
  });
});
