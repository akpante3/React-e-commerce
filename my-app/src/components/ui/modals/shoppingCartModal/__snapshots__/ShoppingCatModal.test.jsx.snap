// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for ShoppingCartModal test for ShoppingCartModal 1`] = `
<ContextProvider
  value={null}
>
  <withRouter(ShoppingCartModal)
    actions={
      Object {
        "generateShoppingId": [Function],
        "getShoppingCartItems": [Function],
      }
    }
    shoppingCartItems={
      Array [
        Object {},
      ]
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;
